### ISSUE TITLE
X-Pack plugin download fails on Windows desktop

### ISSUE THREAD

**Issue by ankitgupta07** [elastic/elasticsearch#24570]
**Describe the feature**:

**Elasticsearch version**: 5.2.2

**Plugins installed**: [X-Pack]

**JVM version** (`java -version`): java version "1.8.0_102"

**OS version** (`uname -a` if on a Unix-like system): Windows 10

**Description of the problem including expected versus actual behavior**: While trying to install X-pack plugin, ran into below error -

```elasticsearch-5.2.2\bin>elasticsearch-plugin.bat install file:\\\<<Path to x-pack-5.2.2.zip>>
\Java\jdk1.8.0_102\bin\java.exe was unexpected at this time.```

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. On windows machine, have your JAVA_HOME path have spaces in the directory path. In my case, JAVA_HOME was set to -

```echo %JAVA_HOME%
C:\Program Files (x86)\Java\jdk1.8.0_102```
 2. Issue the command - elasticsearch-plugin.bat install file:\\\<<Path to x-pack-5.2.2.zip>>
 3. Error - Java\jdk1.8.0_102\bin\java.exe was unexpected at this time.

**Provide logs (if relevant)**:

**Comment by dadoonet** [comment]
I'd try to put java in a directory which name does not contain any special char like () or a space.

**Comment by jasontedor** [comment]
Closed by 3f54ae62648b2f0f71f09e83af89e5ec0479c68f

**Comment by jasontedor** [comment]
Move your Java installation to a directory that contains a single space, or apply the change 3f54ae62648b2f0f71f09e83af89e5ec0479c68f to your local copy of elasticsearch-plugin.bat. This will be fixed in 5.4.1.

**Comment by ankitgupta07** [comment]
Thanks for the fix @jasontedor
